python/python/lance/dataset.py

```diff
@@ -3586,6 +3586,104 @@ def sql(self, sql: str) -> "SqlQueryBuilder":
+    ) -> "DatasetDelta":
+        """
+        Compare changes between two versions of this dataset.
+
+        You must specify either ``compared_against`` (shorthand for comparing the
+        current version against a specific older version) or both ``begin_version``
+        and ``end_version`` for an explicit range.
+
+        Parameters
+        ----------
+        compared_against : int, optional
+            The version to compare the current dataset version against.
+            This is a shorthand for setting ``begin_version=compared_against``
+            and ``end_version=self.version``.
+        begin_version : int, optional
+            The start version (exclusive) for the comparison range.
+            Must be used together with ``end_version``.
+        end_version : int, optional
+            The end version (inclusive) for the comparison range.
+            Must be used together with ``begin_version``.
+
+        Returns
+        -------
+        DatasetDelta
+            An object that can list transactions or stream inserted/updated rows.
+
+        Raises
+        ------
+        ValueError
+            If both ``compared_against`` and version range are specified,
+            or if neither is specified.
+
+        Examples
+        --------
+        .. code-block:: python
+
+            import lance
+            import pyarrow as pa
+
+            # Write initial data (v1)
+            ds = lance.write_dataset(
+                pa.table({"id": [1, 2], "val": ["a", "b"]}),
+                "memory://delta_demo"
+            )
+
+            # Append some data to create v2
+            ds_append = lance.write_dataset(
+                pa.table({"id": [3], "val": ["c"]}),
+                "memory://delta_demo",
+                mode="append"
+            )
+
+            # Compute inserted rows from v1 -> v2 (shorthand)
+            delta = ds_append.delta(compared_against=1)
+            reader = delta.get_inserted_rows()
+            for batch in reader:
+                print(batch)
+
+            # Or using explicit version range
+            delta = ds_append.delta(begin_version=1, end_version=2)
+        """
+        has_compared_against = compared_against is not None
+        has_range = begin_version is not None or end_version is not None
+
+        if has_compared_against and has_range:
+            raise ValueError(
+                "Cannot specify both 'compared_against' and "
+                "'begin_version'/'end_version'. Use one or the other."
+            )
+
+        if not has_compared_against and not has_range:
+            raise ValueError(
+                "Must specify either 'compared_against' or both "
+                "'begin_version' and 'end_version'."
+            )
+
+        if has_range:
+            if begin_version is None or end_version is None:
+                raise ValueError(
+                    "Both 'begin_version' and 'end_version' must be specified together."
+                )
```

Comment:
I mean, this block is not necessary, because it has been checked in Rust:

https://github.com/lance-format/lance/blob/b21872585560cc4538a1f1977d5ac89f799b75c2/rust/lance/src/dataset/delta.rs#L105

And it could be reused for Python and Java?